lib/admin-page.php

```diff
@@ -366,14 +366,44 @@ function classicpress_check_can_migrate() {
 	}
 	echo "\n</p>\n";
 
+	// Check: Core files checksums
+    include_once ABSPATH . 'wp-admin/includes/class-wp-upgrader.php';
+    $upgrader = new Core_Upgrader();
+	if ( !$upgrader->check_files() ) {
+		$preflight_checks['wp_core_files_checksums'] = false;
+		echo "<tr>\n<td>$icon_preflight_fail</td>\n<td>\n";
+	} else {
+		$preflight_checks['wp_core_files_checksums'] = true;
+		echo "<tr>\n<td>$icon_preflight_pass</td>\n<td>\n";
+	}
+	echo "<p>\n";
+	_e(
+		'When migrating to ClassicPress, your WordPress core files inside /wp-admin and /wp-includes will be overwritten and any customisations will be lost.',
+		'switch-to-classicpress'
+	);
+	echo "\n<br>\n";
+	if ( $preflight_checks['wp_core_files_checksums'] ) {
+		_e(
+			'You have not modified any core files.',
+			'switch-to-classicpress'
+		);
+	} else {
+		_e(
+			'It seems you have previously <strong class="cp-emphasis">modified some core files</strong>. You can still migrate to ClassicPress, but be aware that any customisations made previously in those files will be lost.',
+			'switch-to-classicpress'
+		);
+	}
+	echo "\n</p>\n";
+
 	// TODO: Any other checks needed?
 
 	echo "</table>\n";
 
 	if (
 		$preflight_checks['wp_version'] &&
 		$preflight_checks['php_version'] &&
-		$preflight_checks['wp_http_supports_ssl']
+		$preflight_checks['wp_http_supports_ssl'] &&
+		$preflight_checks['wp_core_files_checksums']
```

Comment:
If this check is blocking the migration, then `$preflight_checks['wp_core_files_checksums']` also needs to be verified in `lib/update.php`.

If this check is not going to block the migration, then we need to remove this conditional here.

Reply:
decided to make the checksums non-blocking and added a warn level